jax/_src/numpy/linalg.py

```diff
@@ -469,12 +470,19 @@ def norm(x, ord=None, axis : Union[None, Tuple[int, ...], int] = None,
       # code has slightly different type promotion semantics, so we need a
       # special case too.
       return jnp.sum(jnp.abs(x), axis=axis, keepdims=keepdims)
-    else:
+    elif isinstance(ord, numbers.Number):
       abs_x = jnp.abs(x)
       ord = lax_internal._const(abs_x, ord)
       ord_inv = lax_internal._const(abs_x, 1. / ord)
       out = jnp.sum(abs_x ** ord, axis=axis, keepdims=keepdims)
       return jnp.power(out, ord_inv)
+    else:
```

Comment:
Should we just automatically convert `"inf"` to `jnp.inf`? It might be easier to use.

Reply:
maybe at the top we could do something like:
```python
if isinstance(ord, str):
  ord = float(ord)
```
Would that be sufficient to address the bug?

Reply:
Oh, I see, the desire is to fail for string input. In that case I'd say detect if `ord` is a string and then raise an appropriate `ValueError`

Reply:
Sounds good, I've changed it to do that.